A collection of DAE models (and their ODE equivalent whenever possible)

agarny opened the pull request:
Closes #838.

agarny committed:
Added both the ODE and DAE forms of the the Robertson model.

hsorby commented:
How many models (pairs of models) are you expecting to add to this collection?

agarny commented:
No idea at this stage. There is the Robertson model, which can be seen as our simple test model. Then, there is that Saucerman model that I mentioned in our Zoom call. That one is definitely more involved (from what I remember). It should be added at some point (after having doubled checked its status in PMR).

agarny committed:
Renamed the folders that contain the Robertson model.

agarny committed:
Merge branch 'main' into issue838.

agarny committed:
Merge branch 'main' into issue838

agarny committed:
Merge branch 'main' into issue838.

agarny committed:
Merge branch 'main' into issue838.

agarny committed:
Merge branch 'main' into issue838.

agarny committed:
Merge branch 'main' into issue838.

agarny committed:
Merge branch 'main' into issue838

agarny committed:
Merge branch 'main' into issue838

hsorby committed:
Merge branch 'main' into issue838

agarny committed:
Make sure that MathML elements are properly used.

agarny committed:
Some minor cleaning up.

agarny committed:
Merge branch 'issue1079' into issue1082.

agarny committed:
Merge branch 'issue1079' into issue1082.

agarny committed:
Some minor cleaning up.

agarny committed:
Make sure that MathML elements are properly used.

agarny committed:
Merge branch 'issue1079' into issue1082.

agarny committed:
Merge branch 'main' into issue1082

agarny committed:
Merge branch 'issue1079' into issue1082.

agarny committed:
Merge branch 'issue1079' into issue1082.

agarny committed:
Merge branch 'issue1079' into issue1082.

agarny committed:
Analyser: don't sort variables and equations.

We used to have those calls to `std::sort()` to make the generated code as "pretty" as possible, but it can slow the analysis down quite a bit (by more than 1 second for a "proper" model like the Fabbri one).

agarny committed:
Merge branch 'issue1079' into issue1082.

agarny committed:
Merge branch 'issue1089' of https://github.com/agarny/libcellml into issue1089

agarny committed:
Analyser: don't sort variables and equations.

We used to have those calls to `std::sort()` to make the generated code as "pretty" as possible, but it can slow the analysis down quite a bit (by more than 1 second for a "proper" model like the Fabbri one).

agarny committed:
Merge branch 'issue1079' into issue1082

agarny committed:
Merge branch 'issue1082' into issue1089.

agarny committed:
Don't call ast->type() multiple times.

agarny committed:
Merge branch 'issue1079' into issue1082.

agarny committed:
Merge branch 'issue1082' into issue1089.

agarny committed:
Slight improvement to nonCommentChildNode().

agarny committed:
Slight improvement to our validator.

agarny committed:
Improved branch coverage in nonCommentChildNode().

agarny committed:
Merge branch 'issue1082' into issue1089.

nickerso committed:
Merge branch 'main' into issue1082

agarny committed:
Merge branch 'issue1082' into issue1089.

agarny committed:
Some minor cleaning up.

agarny committed:
Merge branch 'issue1089' into issue838.

agarny commented:
This is now part of PR #942 (see e982f342 which includes the ODE and DAE versions of the Robertson model), so closing as a result.
